src/main/java/org/cbioportal/web/SessionServiceController.java

```diff
@@ -232,6 +244,29 @@ public ResponseEntity<Session> getSession(@PathVariable Session.SessionType type
         }
     }
 
+    /**
+     * This method populates the `virtualStudyData` object with a new set of sample IDs retrieved as the result of executing a query based on virtual study view filters.
+     * It first applies the filters defined within the study view, runs the query to fetch the relevant sample IDs, and then updates the virtualStudyData to reflect these fresh results.
+     * This ensures that the virtual study contains the latest sample IDs.
+     * @param virtualStudyData
+     */
```

Comment:
Nice! Quick question: will this be cached? If yes, how?

Reply:
@pieterlukasse this method is not cached, but `studyViewFilterApplier.apply(virtualStudyData.getStudyViewFilter());` is. I don't think we should cache this method as it'd be cheap transformation.